src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -306,5 +307,74 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
                 return ret;
             }
         }
+        
+        /// <summary>
+        /// Reads an implementation of IDataReader into a DataFrame.
+        /// </summary>
+        /// <param name="reader">DataReader to be read in</param>
+        /// <param name="columnNames">column names (can be empty)</param>
+        /// <param name="dataTypes">column types (can be empty)</param>
+        /// <param name="numberOfRowsToRead">number of rows to read not including the header(if present)</param>
+        /// <param name="addIndexColumn">add one column with the row index</param>
+        /// <returns><see cref="DataFrame"/></returns>
+        public static DataFrame FromDataReader(IDataReader reader, string[] columnNames = null, Type[] dataTypes = null, long numberOfRowsToRead = -1, bool addIndexColumn = false)
```

Comment:
When the PR was opened initially, this is how LoadCsv may have looked. When this PR becomes active again, we should finalize the shape/parameters in this API.